src/basic/fileio.c

```diff
@@ -414,13 +404,8 @@ int read_full_virtual_file(const char *filename, char **ret_contents, size_t *re
                         size = st.st_size;
                         n_retries--;
                 } else {
-                        /* Double the buffer size */
-                        if (size >= READ_FULL_BYTES_MAX)
-                                return -E2BIG;
-                        if (size > READ_FULL_BYTES_MAX / 2 - 1)
-                                size = READ_FULL_BYTES_MAX; /* clamp to max */
-                        else
-                                size = size * 2 + 1; /* Stay always one less than page size, so we malloc evenly */
+                        size = READ_FULL_BYTES_MAX;
```

Comment:
hmm, i am not sure this is really that clearly a good change. sysfs doesn't allow exporting files that are larger than one page for exmaple, so on all of sysfs (which is is probably for more than for any other virtual fs) this will generally heavily over-allocate

Reply:
That's true. So maybe let's optimize this a bit, and if the path starts with /sys, use 4k?

Reply:
Updated like that.

Reply:
Actually, files in /sys/ report a non-zero value, so we'll not overallocate. I reverted to the previous version.